### DISCUSSION TITLE
How to build a custom clip path transition?

### DISCUSSION THREAD

**Discussion by xavierfoucrier** [nolimits4web/swiper#4468] [Q&A] [answered]
Hello everyone!

Pretty new to use SwiperJS in my projects, it is a great slider and well tree-shakeable with webpack.

I am trying to build a **custom "clip path" transition**, but looks like there is no documentation about how to achieve custom transitions. What I am trying to achieve is:

- 1 big slide per view (full screen format)
- only images in all my slides + text over images (like title/subtitle)
- doing a clip-path transition with a circle shape (my main goal)
- being able to position the circle shape where I want on the slide (right bottom, or middle, etc.)
- having a "loopable" slider (being able to go backward/forward and play clip-path transition in both directions)

Thanks a lot for your feedback! :wink:

**Comment by xavierfoucrier** [comment]
@nolimits4web up. Thanks :wink:

**Comment by linonetwo** [comment, marked as answer]
You can write a custom plugin, assign custom css in setTranslate(), see existed plugin for example  https://github.com/nolimits4web/swiper/blob/c7327d41a41ceb5a7685d8ccdace11253836c420/src/components/effect-coverflow/effect-coverflow.js#L5

**Reply by xavierfoucrier** [reply]
Simple as that, OK thanks for the answer! 😉

**Reply by jackkemmish** [reply]
Hey @xavierfoucrier. Did you get anywhere with this, would love to see an example!

**Reply by xavierfoucrier** [reply]
Unfortunately, nothing I could share 😿

My question is quite old, but if I remember well I coded a custom plugin for that.
Follow the way native components are written and you should be able to this.

In addition, search inside new topics, probably some developers already shared their own experience.
Good luck! 😉

**Reply by jackkemmish** [reply]
No worries at all!

Glad to hear you cracked it though. I'll have a snoop around to see if there's any inspiration for what I'm after 🙏 

Thanks for the quick reply!
